### PR TITLE
Fix #12352: replace Mode.ReadComments with -Yread-comments

### LINKED ISSUE #12352: TASTy reflect docstring of Scala library types is `None`
## Compiler version

3.0.0-RC3 and 3.0.1-RC1-bin-SNAPSHOT-nonbootstrapped-git-8957e51 (the version on `master` as of this writing)

## Minimized code

`Test.scala`

```Scala
@main def main(): Unit = {
    println("Quotes: " + getDocString[scala.quoted.Quotes])
}
```

`GetDocString.scala`

```scala
import scala.quoted.*

inline def getDocString[T]: Option[String] = ${ getDocStringImpl[T] }

private def getDocStringImpl[T : Type](using Quotes): Expr[Option[String]] = {
    import quotes.reflect.*
    Expr(TypeRepr.of[T].typeSymbol.docstring)
}
```

`build.sbt`

```scala
val dottyVersion = "3.0.0-RC3"

lazy val root = project
  .in(file("."))
  .settings(
    name := "docstring-none",
    version := "0.1.0",
    scalaVersion := dottyVersion,
    scalacOptions ++= Seq(
      "-Yretain-trees",
      "-Ycook-comments"
    )
  )
```

## Output

```scala
Quotes: None
```

## Expectation

It should print a `Some` containing the documentation of `Quotes`:

https://github.com/lampepfl/dotty/blob/8957e51230b843a7b3cff6f8e0698aad6f1d778e/library/src/scala/quoted/Quotes.scala#L18-L24

## Other info

I picked `Quotes` as an example in the above, but it seems that all Scala library types are returning `None` for their `docstring`. 

It seems that the condition of the `if` statement below is `false`, so the then branch isn't being executed.

https://github.com/lampepfl/dotty/blob/8957e51230b843a7b3cff6f8e0698aad6f1d778e/compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala#L888-L895

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #12352: replace Mode.ReadComments with -YreadComments

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/config/ScalaSettings.scala [hunk 1]
```diff
@@ -178,6 +178,7 @@ class ScalaSettings extends Settings.SettingGroup with CommonScalaSettings {
   val YtestPickler: Setting[Boolean] = BooleanSetting("-Ytest-pickler", "Self-test for pickling functionality; should be used with -Ystop-after:pickler.")
   val YcheckReentrant: Setting[Boolean] = BooleanSetting("-Ycheck-reentrant", "Check that compiled program does not contain vars that can be accessed from a global root.")
   val YdropComments: Setting[Boolean] = BooleanSetting("-Ydrop-comments", "Drop comments when scanning source files.")
+  val YreadComments: Setting[Boolean] = BooleanSetting("-Yread-comments", "Read comments from tasty.")
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/config/ScalaSettings.scala:181]: The user-facing concept is `docs` and not `comment`. We only use comments internally for historical reasons (old mistake) and should avoid leaking this name to the users. At some point, we should fix our internal naming scheme.

```suggestion
  val YreadDocs: Setting[Boolean] = BooleanSetting("-Yread-docs", "Read comments from tasty.")
```

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/config/ScalaSettings.scala:181]: Now I notice that we should also change `YcookComments` should be renamed to `YcookDocs`.

File: compiler/test/dotty/tools/dotc/BootstrappedOnlyCompilationTests.scala [hunk 2]
```diff
@@ -118,6 +118,7 @@ class BootstrappedOnlyCompilationTests {
     aggregateTests(
       compileFilesInDir("tests/run-macros", defaultOptions.and("-Xcheck-macros")),
       compileFilesInDir("tests/run-custom-args/Yretain-trees", defaultOptions and "-Yretain-trees"),
+      compileFilesInDir("tests/run-custom-args/Yread-comments", defaultOptions and "-Yread-comments"),
```
Comment (nicolasstucki) [inline compiler/test/dotty/tools/dotc/BootstrappedOnlyCompilationTests.scala:121]:
```suggestion
      compileFilesInDir("tests/run-custom-args/Yread-docs", defaultOptions and "-Yread-docs"),
```

**Review by nicolasstucki — CHANGES_REQUESTED** [review]

File: compiler/src/dotty/tools/dotc/config/ScalaSettings.scala (hunk 1 above)
Comment (liufengyun) [reply compiler/src/dotty/tools/dotc/config/ScalaSettings.scala]: Good catch, now it aligns with existing naming.

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**